doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -1906,9 +1906,38 @@ Numbers and strings
 
 Numbers and strings have no predefined semantics in the calculus. They are
 merely notations that can be bound to objects through the notation mechanism.
-Initially, numbers are bound to Peano’s representation of natural
+Initially, numbers are bound to :n:`nat`, Peano’s representation of natural
 numbers (see :ref:`datatypes`).
 
+Valid Ranges for Numbers
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, numbers are interpreted as :n:`nat`\s, which is a unary representation.
+For example, :n:`3` is represented as `(S (S (S O)))`.  Computing with large :n:`nat`\s
+can lead to stack overflows.  You can explicitly specify a different interpretation,
+which will avoid this problem.  For example :n:`3%Z` represents the number as an integer.
+See :ref:`Scopes` and :n:`@term_scope`.
+
+.. todo PR: what is the specific type for the integer?  Is there a range limitation?
+
+   .. todo PR When do the 63-bit range limits apply?
```

Comment:
This limit is about the parsing and interpretation of syntax, and has nothing to do with stack overflows